test/utils/asserts.hpp

```diff
@@ -100,6 +100,8 @@ MATCHER_P(CResult, value, "")  // NOLINT(readability-redundant-string-init)
             return arg.value.f64 == fizzy::test::FP{value};
         }
     }
+    else if constexpr (std::is_same_v<value_type, uint32_t>)
```

Comment:
I think this should also include `int32_t`?

Reply:
As changed in #687, only `_u32` and `_u64` are accepted for `CResult()`.

Reply:
Hm, writing a test expecting a negative value will be a bit awkward then, istn't it?

Reply:
`-1_u32`.

Reply:
But isn't it equivalent to `-uint32_t{1}`, so signed value?

Reply:
I'm confused about why `Result` and `CResult` are different in this regard.
Is `Result` allowing signed values because otherwise there're many tests to change?

Reply:
> But isn't it equivalent to `-uint32_t{1}`, so signed value?

It is, but it stays `uint32_t` type.



> I'm confused about why `Result` and `CResult` are different in this regard.
> Is `Result` allowing signed values because otherwise there're many tests to change?

The `Result()` requires type information. I restricted values to `CResult()` more because the `EXPECT_THAT(Value{0x100000000_u64}, CResult(0_u32))` is a match.

Reply:
> `EXPECT_THAT(Value{0x100000000_u64}, CResult(0_u32))` is a match.

How does restricting to only unsigned values help this?

Reply:
You can't put `CResult(0)` so you need to thing if it should be `_u32` or `_u64`. This also affects existing tests.

Reply:
I see. And what is the disadvantage of making `Result()` accept only unsigned expected types?

Reply:
It may be good idea, but this requires 176 additional test changes. So I went with adding type checking instead. Can be done later if desired.

We can also require type in `CResult` or even merge these two. But I believe there is additional research needed here. E.g. we should try combining matchers: `Result: Not(Traps) && HasResult && ResultValueIs(x)`.

Reply:
I see, that's what I guessed in the beginning, too many tests to change to make `Result` accept only unsigned,